Runtime/Loggers/AbstractLogger.cs

```diff
@@ -0,0 +1,29 @@
+// Copyright (c) 2023-2024 DeNA Co., Ltd.
+// This software is released under the MIT License.
+
+using System;
+using UnityEngine;
+
+namespace DeNA.Anjin.Loggers
+{
+    /// <summary>
+    /// Abstract logger settings used for autopilot.
+    /// </summary>
+    public abstract class AbstractLogger : ScriptableObject, IDisposable
+    {
+#if UNITY_EDITOR
+        /// <summary>
+        /// Description about this agent instance.
+        /// </summary>
+        [Multiline] public string description;
+#endif
+
+        /// <summary>
+        /// Logger implementation used for autopilot.
+        /// </summary>
+        public abstract ILogger LoggerImpl { get; }
```

Comment:
How about this suggestion?

```suggestion
    public abstract class AbstractLoggerSettings : ScriptableObject, IDisposable
    {
#if UNITY_EDITOR
        /// <summary>
        /// Description about this agent instance.
        /// </summary>
        [Multiline] public string description;
#endif

        /// <summary>
        /// Logger implementation used for autopilot.
        /// </summary>
        public abstract ILogger Logger { get; }
```

Reply:
Thanks, I'm a little hesitant to name it "settings", but I don't have another idea...